nx_parallel/algorithms/centrality/tests/test_betweenness_centrality.py

```diff
@@ -0,0 +1,33 @@
+import networkx as nx
+import nx_parallel as nxp
+
+
+def test_betweenness_centrality_get_chunks():
+    def get_chunk(nodes, num_chunks):
+        nodes_ebc = {i: 0 for i in nodes}
+        for i in ebc:
+            nodes_ebc[i[0]] += ebc[i]
+            nodes_ebc[i[1]] += ebc[i]
+
+        sorted_ebc = sorted(nodes_ebc.items(), key=lambda x: x[1])
+
+        chunks_with_avg = [[] for _ in range(num_chunks)]
+        chunk_sums = [0] * num_chunks
+
+        # Assign values to chunks based on balancing the sums
+        for key, value in sorted_ebc:
+            # Find the chunk with the smallest sum and add the value to it
+            min_chunk = min(enumerate(chunk_sums), key=lambda x: x[1])[0]
+            chunks_with_avg[min_chunk].append((key, value))
+            chunk_sums[min_chunk] += value
+
+        chunks = [[j[0] for j in i] for i in chunks_with_avg]
+
+        return chunks
+
+    G = nx.fast_gnp_random_graph(50, 0.5, seed=5, directed=False)
+    ebc = nx.edge_betweenness_centrality(G)
+    par_bc = nxp.betweenness_centrality(G, get_chunks=get_chunk)  # smoke test
+    bc = nx.betweenness_centrality(G)
+    for i in range(20):
+        assert round(par_bc[i], 15) == round(bc[i], 15)
```

Comment:
The test was failing because of the last few decimal places for 5-6 nodes, so I used `round()`. I don't think if this is happening bcoz of `get_chunk`. I'm looking into this. Let me know if you know why this might be happening.

Reply:
That is almost certainly because of round-off error when computing/storing the numbers. Floating point values (double precision) have round-off errors typically around 10^-16, and so the rounding to 15 decimals aligns with that. In fact round-off error often becomes larger than that over more than one arithmetic operation, so it is nice that these errors are as small as they are. (round-off is different for different machines, for different orders of the operations, etc.. They aren't random, but they aren't predictable. :)

Reply:
Another way to check that floating point numbers are close is [`math.isclose` ](https://docs.python.org/3/library/math.html#math.isclose) or the numpy version [`numpy.isclose`](https://numpy.org/doc/stable/reference/generated/numpy.isclose.html).  They handle the edge cases where rounding gets tricky (relative error vs absolute error, etc).